napari/conftest.py

```diff
@@ -450,6 +451,38 @@ def single_threaded_executor():
     executor.shutdown()
 
 
+@pytest.fixture()
+def mock_console(request):
+    """Mock the qtconsole to avoid starting an interactive IPython session.
+    In-process IPython kernels can interfere with other tests and are difficult
+    (impossible?) to shutdown.
+
+    This fixture is configured to be applied automatically to tests unless they
+    use the `enable_console` marker. It's not autouse to avoid use on headless
+    tests (without Qt); instead it's enabled in `pytest_runtest_setup`.
```

Comment:
What does "instead, it's enabled in `pytest_runtest_setup`" mean sorry?

Reply:
Ash has headless test that runs tests that do not depends on qt we cannot enable this fixture for all test as it will cause the crashing of the test run. 

We met this problem first when creating fixtures for detect leaking timers, threads etc. Then I wrote  `pytest_runtest_setup` function (placed in `conftest.py`) which is called during setup of each test function by pytest itself. This function checks the presence of `qapp` fixture that indicate that the test has `qtbot`, or `qapp` fixture which means that this test depends on Qt.